i6_models/parts/frontend/protocol.py

```diff
@@ -0,0 +1,23 @@
+import torch
+import torch.nn as nn
+from typing import Protocol, Tuple
+
+
+class BaseFrontendInterfaceV1(Protocol):
+    def forward(self, tensor: torch.Tensor, sequence_mask: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        :param tensor: input tensor of shape [B,T,F]
+        :param sequence_mask: masking tensor of shape [B,T], contains length information of the sequences
+        :return: torch.Tensor of shape [B,T',F']
+        """
+        raise NotImplementedError
+
+
+class FrontendInterfaceV1(BaseFrontendInterfaceV1, nn.Module):
```

Comment:
I'm not sure it works that you can combine a `Protocol` and the condition that sth is derived from `nn.Module`. I think this does not work.

We had a similar discussion here:
https://github.com/rwth-i6/i6_models/pull/21#discussion_r1245025446
https://github.com/rwth-i6/i6_models/pull/21#discussion_r1246450812

I'm not sure though what would be the best option here. I think we can only have one: either the `Protocol`, or saying it must derive from `nn.Module`.

Maybe the protocol, the interface can just be informal, i.e. we define it somewhere in the documentation, but otherwise it is not enforced. Although you would get an error obviously when you pass some incorrect module.